File: darkwater-core/src/main/java/kiinse/plugins/darkwaterapi/core/files/locale/Locale.java
package kiinse.plugins.darkwaterapi.core.files.locale;

import kiinse.plugins.darkwaterapi.api.files.locale.PlayerLocale;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public class Locale extends PlayerLocale {

    private final String value;

    protected Locale(@NotNull String str) {
        value = str;
    }

    @Override
    public @NotNull String toString() {
        return value;
    }

    @Override
    public boolean equals(@NotNull PlayerLocale playerLocale) {
        return Objects.equals(this.toString(), playerLocale.toString());
    }
}

File: core/src/main/kotlin/kiinse/me/plugins/darkwaterapi/core/files/locale/Locale.kt
package kiinse.me.plugins.darkwaterapi.core.files.locale

import kiinse.me.plugins.darkwaterapi.api.files.locale.PlayerLocale

class Locale(private val value: String) : PlayerLocale() {

    override fun toString(): String {
        return value
    }

    override fun equals(other: Any?): Boolean {
        return other != null && other is PlayerLocale && this.hashCode() == other.hashCode()
    }

    override fun hashCode(): Int {
        return toString().hashCode()
    }
}